<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="alexander-gysin (alexander.gysin@usz.ch)" />


<title>Olink-GB</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">omics-training</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Olink-GB.html">Olink Analysis</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/alexander-gysin/omics-training">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Olink-GB</h1>
<h4 class="author">alexander-gysin (<a
href="mailto:alexander.gysin@usz.ch"
class="email">alexander.gysin@usz.ch</a>)</h4>
<h4 class="date">22 February, 2026</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-02-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>omics-training/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20260210code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20260210)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20260210code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20260210)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomalexandergysinomicstrainingtree4a2c638f2fc0044623fff1b632836ccd6e060e7atargetblank4a2c638a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/alexander-gysin/omics-training/tree/4a2c638f2fc0044623fff1b632836ccd6e060e7a" target="_blank">4a2c638</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomalexandergysinomicstrainingtree4a2c638f2fc0044623fff1b632836ccd6e060e7atargetblank4a2c638a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/alexander-gysin/omics-training/tree/4a2c638f2fc0044623fff1b632836ccd6e060e7a" target="_blank">4a2c638</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/Olink-GB/

Untracked files:
    Untracked:  code/Olink-GB/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/_site.yml
    Deleted:    analysis/first-analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Olink-GB.Rmd</code>) and HTML
(<code>docs/Olink-GB.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/4a2c638f2fc0044623fff1b632836ccd6e060e7a/analysis/Olink-GB.Rmd" target="_blank">4a2c638</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-22
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/9d5e0171c682bbe751613e473e6f589961538be4/docs/Olink-GB.html" target="_blank">9d5e017</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/664f17cf11034dc6bf4ccc26b06002345e48a474/analysis/Olink-GB.Rmd" target="_blank">664f17c</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-22
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/cb62f2a4767ff709d6940b5741d89ebe69302574/docs/Olink-GB.html" target="_blank">cb62f2a</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/69b5ab2834e1c8dabd0c2d50ba5b7eb541151231/analysis/Olink-GB.Rmd" target="_blank">69b5ab2</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-22
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/c37d3e1c9ac3013b8b72e495b3693c63f6f45f36/docs/Olink-GB.html" target="_blank">c37d3e1</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/e95c6b96a5a170fd1af9bcf45cd27ad6dd56d2c7/analysis/Olink-GB.Rmd" target="_blank">e95c6b9</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-20
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/942320ddda713a33bee70d6f7a06c8c501659866/docs/Olink-GB.html" target="_blank">942320d</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/75a9e0cca720a5f632215d7a232f6d7d373c2686/analysis/Olink-GB.Rmd" target="_blank">75a9e0c</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-20
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/1e15a4ce30f6a3855a6560624fd3bdb447dc1eb2/docs/Olink-GB.html" target="_blank">1e15a4c</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/10e9eb58470c22934f1af49b1dde9531c532e516/analysis/Olink-GB.Rmd" target="_blank">10e9eb5</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-19
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/558ae77c852b60771a5ce16de896a786bab48a63/docs/Olink-GB.html" target="_blank">558ae77</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/0eada49f4656cef4fa3b19f04895ec010bcd1903/analysis/Olink-GB.Rmd" target="_blank">0eada49</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/cbd4b89948a69630b8cdee6f3deb815e89b94354/docs/Olink-GB.html" target="_blank">cbd4b89</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/f9cf585f1a3595cc77adea9066b844b8d42e2f64/analysis/Olink-GB.Rmd" target="_blank">f9cf585</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/f89194a64ee261f9e0c2b0c260231017243b71f7/docs/Olink-GB.html" target="_blank">f89194a</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/f29a5c64c6d0c36e713bbece8c6a90699037bc67/analysis/Olink-GB.Rmd" target="_blank">f29a5c6</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/5f2d281035502be1ea201bebe45f8e7cc20f3335/docs/Olink-GB.html" target="_blank">5f2d281</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/54609d1dd9114cc8b465e272da80a0ad698802f6/analysis/Olink-GB.Rmd" target="_blank">54609d1</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/9ba78e7f24bd98107a5e62a732537c23398a02c6/docs/Olink-GB.html" target="_blank">9ba78e7</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/48ac657607c15ead3162113aec9a8282d3f4072d/analysis/Olink-GB.Rmd" target="_blank">48ac657</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/cb4c6c9fa1e1879d68971ef4d5a5547b303dd84a/docs/Olink-GB.html" target="_blank">cb4c6c9</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/632763f2ea5f491d0af0fb371f7b885ce2effa59/analysis/Olink-GB.Rmd" target="_blank">632763f</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Hide all except certain code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/a45865e9f58830693dfd08e2de0229606879372e/docs/Olink-GB.html" target="_blank">a45865e</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/7ac5b7069b5bb3e8b41a4a9b9db7eff5c8e5973a/docs/Olink-GB.html" target="_blank">7ac5b70</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/8fd6e5e660effa56918a26a606bebc297a9f3acc/analysis/Olink-GB.Rmd" target="_blank">8fd6e5e</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/8635b5ce733aebf9d67222c435e28d98252cd87c/docs/Olink-GB.html" target="_blank">8635b5c</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/a5e69971d098f0f0523d3dbd6fd1ca87f406e5ab/analysis/Olink-GB.Rmd" target="_blank">a5e6997</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/8fc46d7030714fd488cb2247e34066b19133b4da/docs/Olink-GB.html" target="_blank">8fc46d7</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/c583496bc28a96d12873bad42a01ebfc03bb2490/analysis/Olink-GB.Rmd" target="_blank">c583496</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/a00cade57b936659b8f9850db0eeb521f55829c1/docs/Olink-GB.html" target="_blank">a00cade</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/608c7931d26de437762a9679acabd4d4c034e506/analysis/Olink-GB.Rmd" target="_blank">608c793</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/7b8d3a56178290dcf33f811842a2ad3bb0144298/docs/Olink-GB.html" target="_blank">7b8d3a5</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/f2390c0024f04f88c8b8cc53cb2137884de9454e/analysis/Olink-GB.Rmd" target="_blank">f2390c0</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This serves as training to analyze Olink data</p>
<div id="setup" class="section level2 tabset">
<h2 class="tabset">Setup</h2>
<div id="directories" class="section level3">
<h3>Directories</h3>
<pre class="r"><code># Get current file name to make folder
current_file &lt;- &quot;Olink-GB&quot;
#Output paths
output_dir_data &lt;- here::here(&quot;output&quot;, current_file)
if (!dir.exists(output_dir_data)) dir.create(output_dir_data)

if (!dir.exists(here::here(&quot;docs&quot;, &quot;figure&quot;))) dir.create(here::here(&quot;docs&quot;, &quot;figure&quot;))</code></pre>
</div>
<div id="knitr-configuration" class="section level3">
<h3>knitr configuration</h3>
<pre class="r"><code>knitr::opts_chunk$set(
  warning = FALSE,
  message = FALSE
)</code></pre>
</div>
<div id="packages" class="section level3">
<h3>Packages</h3>
<pre class="r"><code># Utility Packages
library(tidyverse)  # Contains dplyr (data manipulation), readr, tibble (modern df), ggplot2 and more
library(here)       # Uses the project root for file paths; stops &quot;File not found&quot; errors on other PCs
library(magrittr)   # Provides the pipe operator (%&gt;%) to make your code readable
library(RColorBrewer) # Provides professional, color-blind friendly palettes for your plots
library(readxl)     # Used to load excel files

# Differential Expression &amp; PCA Packages
library(limma)            # The gold standard for linear modeling and DEA (works for Olink/Proteomics)
library(stats)            # Built-in R package; provides prcomp() for PCA (no need to load explicitly)
library(FactoMineR)       # Multi Factor Analysis for multi-omics, calculates the contribution to the PCA
library(factoextra)       # PCA Visualization

# Protein Protein interaction
library(rbioapi)          # Allows to access the STRING database via API

# Gene Set Enrichment Analysis
library(msigdbr)          #Allows to download GeneSets
library(clusterProfiler)  #Performs the GSEA provides Statistical correlation, visualization and some flexibility (can handle GO and KEGG)
library(enrichplot)
library(ggridges)

# Visualization Packages
library(EnhancedVolcano)  # Specifically designed for high-quality, easy volcano plots
library(pheatmap)         # Quick and clean heatmaps to visualize clusters after PCA
library(patchwork)        # Essential for combining multiple plots (e.g., PCA next to a Volcano plot)
library(kableExtra)       #better display of tables in the knitted document</code></pre>
</div>
<div id="ggplot-global-theme" class="section level3">
<h3>ggplot global theme</h3>
<pre class="r"><code># Set the default ggplot theme globally
theme_set(
  theme_bw(base_size = 12) + 
  theme(
    panel.grid.minor = element_blank(),
    plot.title = element_text(face = &quot;bold&quot;, size = 14),
    strip.background = element_rect(fill = &quot;grey95&quot;)
  )
)</code></pre>
</div>
</div>
</div>
<div id="data-preparation-qc" class="section level1">
<h1>Data preparation &amp; QC</h1>
<div id="initialization" class="section level2">
<h2>Initialization</h2>
<div id="read-data" class="section level3">
<h3>Read data</h3>
<pre class="r"><code>Olink_data &lt;- read.csv(here::here(&quot;data&quot;, &quot;quant.csv&quot;), sep = &quot;;&quot;)
Clinical_data &lt;- read_excel(here::here(&quot;data&quot;, &quot;Metadata_Olink_PL.xlsx&quot;))</code></pre>
</div>
<div id="define-color-palette" class="section level3">
<h3>Define color palette</h3>
<pre class="r"><code>my_palette &lt;- colorRampPalette(c(&quot;navy&quot;, &quot;whitesmoke&quot;, &quot;firebrick3&quot;))(100)</code></pre>
</div>
</div>
<div id="data-clean-up" class="section level2">
<h2>Data clean-up</h2>
<div id="formating-the-olink-file" class="section level3">
<h3>Formating the Olink file</h3>
<p>Protein names to rownames in the Olink file</p>
<pre class="r"><code>Olink_data &lt;- Olink_data %&gt;%
  column_to_rownames(&quot;SampleID&quot;) # Replace &quot;ProteinID&quot; with your actual column name</code></pre>
</div>
<div id="order-olink-data-according-to-metadata" class="section level3">
<h3>Order Olink data according to metadata</h3>
<pre class="r"><code>Olink_data &lt;- Olink_data[, Clinical_data$Barcode]

# Check if column names match metadata barcodes exactly
stopifnot(all(colnames(Olink_data) == Clinical_data$Barcode))</code></pre>
</div>
<div id="remove-the-olink_id-from-the-rownames" class="section level3">
<h3>Remove the Olink_ID from the rownames</h3>
<pre class="r"><code>rownames(Olink_data) &lt;- str_remove(rownames(Olink_data), &quot;_.*&quot;)</code></pre>
</div>
<div id="save-processed-data" class="section level3">
<h3>Save Processed Data</h3>
<pre class="r"><code># These are your core objects for all future analysis
saveRDS(Olink_data, file = file.path(output_dir_data, &quot;Olink_clean.rds&quot;))
saveRDS(Clinical_data, file = file.path(output_dir_data, &quot;Clinical_clean.rds&quot;))</code></pre>
</div>
<div id="check-for-missing-values-in-the-olink-file"
class="section level3">
<h3>Check for missing values in the Olink file</h3>
<pre class="r"><code>sum(is.na(Olink_data))</code></pre>
<pre><code>[1] 0</code></pre>
</div>
</div>
</div>
<div id="results" class="section level1">
<h1>Results</h1>
<div id="pca-analysis" class="section level2">
<h2>PCA Analysis</h2>
<div id="pca-analysis-1" class="section level3">
<h3>PCA Analysis</h3>
<pre class="r"><code># Note: PCA functions usually expect samples in ROWS, so we transpose with t()
pca_res &lt;- PCA(t(Olink_data), graph = FALSE)</code></pre>
</div>
</div>
<div id="pca-visualization" class="section level2">
<h2>PCA visualization</h2>
<div id="plotting-eigenvalues" class="section level3">
<h3>Plotting Eigenvalues</h3>
<pre class="r"><code>p_scree &lt;- fviz_eig(pca_res, addlabels = TRUE, barfill = &quot;navy&quot;, barcolor = &quot;navy&quot;)
p_scree</code></pre>
<p><img src="figure/Olink-GB.Rmd/eigenvalue_vis-1.png" width="672" style="display: block; margin: auto;" />
&gt; <strong>Key Finding:</strong> The first two principal components
capture a substantial portion of the total proteomic variance,
confirming that the high-dimensional Olink data can be effectively
simplified into a 2D space for group comparisons.</p>
<ul>
<li><strong>How to interpret this plot:</strong> Each bar represents a
Principal Component (PC). The height shows the percentage of total
variance explained by that component.</li>
<li><strong>A Scree Plot can show you:</strong> How “structured” your
data is. If the first few bars are tall, it means a few dominant
biological signals are driving the differences between your
samples.</li>
<li><strong>Selection Criteria:</strong> This plot includes the variance
distribution for all proteins measured in the panel.</li>
</ul>
</div>
<div id="plotting-pca-by-group" class="section level3">
<h3>Plotting PCA by Group</h3>
<pre class="r fold-hide"><code># 2. PCA Plot colored by Group
p_pca &lt;- fviz_pca_ind(pca_res,
             label = &quot;none&quot;, # Hide sample names for clarity
             habillage = as.factor(Clinical_data$Group), 
             palette = c(&quot;indianred&quot;, &quot;navy&quot;),
             addEllipses = TRUE, 
             invisible = &quot;quali&quot;) +
             theme_minimal() +
             labs(title = &quot;Proteomics PCA&quot;)
p_pca</code></pre>
<p><img src="figure/Olink-GB.Rmd/PCA_Plot_by_group-1.png" width="672" style="display: block; margin: auto;" />
&gt; <strong>Key Finding:</strong> The protein profiles of SSc-PAH and
SSc-noPAH patients show a visible shift along the primary axes of
variance; however, the partial overlap of the 95% confidence ellipses
indicates that the groups differ moderately rather than exhibiting
entirely distinct proteomic signatures.</p>
<ul>
<li><strong>How to interpret this plot:</strong> Each point represents a
patient sample. The spatial distance between points reflects the
similarity of their overall protein profiles. The shaded “ellipses”
represent the 95% confidence intervals for each group.</li>
<li><strong>A PCA Plot can show you:</strong> If the clinical groups
naturally “cluster” based on protein expression. Overlapping ellipses
suggest high biological variability or a spectrum of disease states,
while distinct clusters indicate a strong, group-specific protein
signature.</li>
<li><strong>Selection Criteria:</strong> This visualization utilizes all
proteins in the filtered dataset to calculate coordinates and includes
all patient samples.</li>
</ul>
</div>
<div
id="visualizing-contribution-by-protein-in-a-simple-contribution-plot"
class="section level3">
<h3>Visualizing contribution by protein in a simple contribution
plot</h3>
<pre class="r fold-hide"><code># manually set a common limit
common_limit &lt;- 0.5
# Visualize the top 10 proteins contributing to PC1
p_top_proteins_PD1 &lt;- fviz_contrib(pca_res, choice = &quot;var&quot;, axes = 1, top = 10, fill = &quot;navy&quot;, color = &quot;navy&quot;, sort.val = &quot;asc&quot;) +
  theme_minimal() +
  labs(title = &quot;Top 10 Proteins Contributing to PC1&quot;,
       x = &quot;Proteins&quot;,
       y = &quot;Contribution [%]&quot;)+
  coord_flip(ylim = c(0, common_limit)) # Flips the plot and sets the scale

# Visualize the top 10 proteins contributing to PC2
p_top_proteins_PD2 &lt;- fviz_contrib(pca_res, choice = &quot;var&quot;, axes = 2, top = 10, fill = &quot;navy&quot;, color = &quot;navy&quot;, sort.val = &quot;asc&quot;) +
  theme_minimal() +
  labs(title = &quot;Top 10 Proteins Contributing to PC2&quot;,
       x = &quot;Proteins&quot;,
       y = &quot;Contribution [%]&quot;)+
  coord_flip(ylim = c(0, common_limit)) # Flips the plot and sets the scale

# / displays the plots in vertical order
p_top_proteins_PD1 / p_top_proteins_PD2</code></pre>
<p><img src="figure/Olink-GB.Rmd/top_10_proteins-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r fold-hide"><code>#alternatively flip the plot horizontally to display more bars </code></pre>
<blockquote>
<p><strong>Key Finding:</strong> A specific subset of proteins
contributes disproportionately to the variance captured by PC1 and PC2,
identifying them as the primary drivers of the differences observed in
the PCA map.</p>
</blockquote>
<ul>
<li><strong>How to interpret this plot:</strong> The y-axis shows the
percentage of a component’s variance attributed to a single protein. The
dashed line represents the “average” expected contribution; any protein
crossing this line is a significant driver of that axis.</li>
<li><strong>A Contribution Plot can show you:</strong> Which specific
proteins are responsible for the separation of your groups. Proteins in
PC1 are likely the strongest differentiators between SSc-PAH and
SSc-noPAH.</li>
<li><strong>Selection Criteria:</strong> The top 10 proteins with the
highest contribution scores for each axis are displayed.</li>
</ul>
</div>
<div id="visualizing-contribution-by-protein-in-a-loading-plot"
class="section level3">
<h3>Visualizing contribution by protein in a loading plot</h3>
<pre class="r fold-hide"><code># Variable Correlation Plot
p_loading &lt;- fviz_pca_var(pca_res, 
                          # Only show the top 20 proteins by &#39;contribution&#39; 
                          # (This keeps the plot clean)
                          select.var = list(contrib = 20), 
                          
                          # Color the proteins by their contribution level
                          col.var = &quot;contrib&quot;, 
                          gradient.cols = c(&quot;navy&quot;, &quot;white&quot;, &quot;indianred&quot;),
                          
                          # Avoid text labels overlapping
                          repel = TRUE, 
                          
                          # Visual aesthetics
                          alpha.var = &quot;contrib&quot;) + 
  theme_minimal() +
  labs(title = &quot;Loading Plot: Top 20 Proteins&quot;,
       subtitle = &quot;Direction shows correlation with PC1 (x) and PC2 (y)&quot;)
p_loading</code></pre>
<p><img src="figure/Olink-GB.Rmd/loading_plot-1.png" width="672" style="display: block; margin: auto;" />
&gt; <strong>Key Finding:</strong> The strong correlation between the
top 20 proteins (vectors pointing in similar directions) suggests they
participate in a shared biological process that characterizes the
SSc-PAH phenotype.</p>
<ul>
<li><strong>How to interpret this plot:</strong> The arrows (vectors)
show how proteins correlate with the PCA axes. Arrows pointing in the
same direction represent highly correlated proteins. The length of the
arrow indicates the strength of the protein’s contribution.</li>
<li><strong>A Loading Plot can show you:</strong> The relationship
between individual proteins and the sample groupings.</li>
<li><strong>Selection Criteria:</strong> The top 20 proteins by
contribution scores are selected to ensure plot legibility.</li>
</ul>
<p>!! plot is not very informative and it seems to labelled redundantly
(contribution two times)</p>
</div>
</div>
<div id="differential-expression-analysis"
class="section level2 tabset">
<h2 class="tabset">Differential Expression Analysis</h2>
<div id="create-the-design-matrix" class="section level3">
<h3>Create the Design Matrix</h3>
<pre class="r"><code># 1. Clean the &#39;Group&#39; column in your metadata directly
# This replaces hyphens/spaces with dots (e.g., SSc-PAH -&gt; SSc.PAH)
Clinical_data$Group_Clean &lt;- make.names(Clinical_data$Group)

# 2. Re-create the group_factor from the clean column
group_factor &lt;- as.factor(Clinical_data$Group_Clean)

# 3. Build the design matrix
design &lt;- model.matrix(~ 0 + group_factor)

# 4. Rename columns to match the clean levels (SSc.PAH, SSc.noPAH)
colnames(design) &lt;- levels(group_factor)</code></pre>
</div>
<div id="define-the-comparison" class="section level3">
<h3>Define the comparison</h3>
<pre class="r"><code># 5. Define the contrast
# Syntax: makeContrasts(Group1 - Group2, levels = design)
contrast.matrix &lt;- makeContrasts(SSc.PAH - SSc.noPAH, levels = design)</code></pre>
</div>
<div id="analysis" class="section level3">
<h3>Analysis</h3>
<pre class="r"><code># Fit the model
fit &lt;- lmFit(Olink_data, design)
fit2 &lt;- contrasts.fit(fit, contrast.matrix)
fit2 &lt;- eBayes(fit2)

# 1. Get the results from limma
de_results &lt;- topTable(fit2, coef = 1, number = Inf, sort.by = &quot;P&quot;) %&gt;%
  rownames_to_column(&quot;Protein&quot;)

# 2. CREATE THE MISSING COLUMN (The Fix)
de_results &lt;- de_results %&gt;%
  mutate(Significance = case_when(
    adj.P.Val &lt; 0.05 &amp; logFC &gt; 0.5  ~ &quot;Up-regulated&quot;,
    adj.P.Val &lt; 0.05 &amp; logFC &lt; -0.5 ~ &quot;Down-regulated&quot;,
    TRUE                          ~ &quot;Not Significant&quot;
  ))</code></pre>
</div>
<div id="save-dea-results" class="section level3">
<h3>Save DEA Results</h3>
<pre class="r"><code># Save the full results table
write.csv(de_results, file = file.path(output_dir_data, &quot;DE_results_SSc_PAH.csv&quot;), row.names = FALSE)

# Also save as RDS to preserve variable types (like the &#39;Significance&#39; factor)
saveRDS(de_results, file = file.path(output_dir_data, &quot;DE_results_SSc_PAH.rds&quot;))</code></pre>
</div>
</div>
<div id="visualization-of-the-differential-analysis"
class="section level2">
<h2>Visualization of the Differential Analysis</h2>
<div id="volcanoplot-preparation" class="section level3">
<h3>Volcanoplot preparation</h3>
<pre class="r fold-hide"><code># --- 1. DATA PREPARATION ---
# Calculate the exact raw p-value that hits the FDR 0.05 threshold
p_cutoff &lt;- de_results %&gt;% 
  filter(adj.P.Val &lt; 0.05) %&gt;% 
  pull(P.Value) %&gt;% 
  max()

# Create a &#39;Score&#39; to identify the &quot;top corners&quot; (High FC * High Significance)
# This ensures we label the most extreme points in the top-left and top-right.
de_results &lt;- de_results %&gt;%
  mutate(label_score = abs(logFC) * -log10(P.Value))

# Identify top 10 Up and top 10 Down based on this combined score
top_up &lt;- de_results %&gt;%
  filter(Significance == &quot;Up-regulated&quot;) %&gt;%
  slice_max(order_by = label_score, n = 10)

top_down &lt;- de_results %&gt;%
  filter(Significance == &quot;Down-regulated&quot;) %&gt;%
  slice_min(order_by = logFC, n = 10) # Using logFC here ensures we stay on the left side

# Re-selecting top_down based on score but ensuring they are actually down-regulated
top_down &lt;- de_results %&gt;%
  filter(Significance == &quot;Down-regulated&quot;) %&gt;%
  slice_max(order_by = label_score, n = 10)

top_labels &lt;- bind_rows(top_up, top_down)</code></pre>
</div>
<div id="volcano-plot" class="section level3">
<h3>Volcano plot</h3>
<pre class="r fold-hide"><code># --- 2. THE FINAL PLOT ---
ggplot(de_results, aes(x = logFC, y = -log10(P.Value), color = Significance)) +
  geom_point(alpha = 0.4, size = 1.5) +
  scale_color_manual(values = c(&quot;Down-regulated&quot; = &quot;navy&quot;, 
                                &quot;Not Significant&quot; = &quot;grey85&quot;, 
                                &quot;Up-regulated&quot; = &quot;indianred&quot;)) +
  
  # Clean, small text labels
  geom_text_repel(data = top_labels, 
                  aes(label = Protein),
                  size = 2.5,
                  color = &quot;black&quot;,
                  box.padding = 0.5,
                  point.padding = 0.2,
                  min.segment.length = 0,
                  segment.size = 0.2,
                  segment.color = &quot;grey30&quot;,
                  max.overlaps = Inf) +
  
  theme_minimal() +
  labs(title = &quot;SSc.PAH vs SSc.noPAH: Top Hits&quot;,
       subtitle = &quot;Labeled by multiplied Significance and Fold Change&quot;,
       x = &quot;Log2 Fold Change&quot;,
       y = &quot;-log10(Raw P-value)&quot;) +
  
  # Guidelines
  geom_vline(xintercept = c(-0.5, 0.5), linetype = &quot;dotted&quot;, color = &quot;grey40&quot;) +
  geom_hline(yintercept = -log10(p_cutoff), linetype = &quot;dashed&quot;, color = &quot;grey40&quot;, alpha = 0.7) +
  
  # Move FDR Label to the far LEFT to avoid crashing with the &quot;Up-regulated&quot; top corner
  annotate(&quot;label&quot;, 
           x = min(de_results$logFC), 
           y = -log10(p_cutoff), 
           label = &quot;FDR &lt; 0.05&quot;, 
           hjust = 0.2,             # Align to the left of the x-coordinate
           vjust = 1.4, 
           size = 2.5, 
           color = &quot;black&quot;, 
           fill = &quot;white&quot;,        # White background prevents &quot;crashing&quot; with grid lines
           label.size = NA,       # Removes the box border
           fontface = &quot;plain&quot;)</code></pre>
<p><img src="figure/Olink-GB.Rmd/volcano_plotting-1.png" width="672" style="display: block; margin: auto;" />
&gt; <strong>Key Finding:</strong> Several proteins demonstrate both
high statistical significance and substantial fold changes, identifying
them as robust candidates for the biological differentiation of
SSc-PAH.</p>
<ul>
<li><strong>How to interpret this plot:</strong> The x-axis represents
the magnitude of change (Log2 Fold Change), and the y-axis represents
statistical significance (*-log10 P-value). Points in the top-right
(red) are upregulated in SSc-PAH, while points in the top-left (blue)
are downregulated.</li>
<li><strong>A Volcano Plot can show you:</strong> The most “robust”
biological hits at a glance. It helps distinguish between proteins that
change drastically in concentration versus those that change
consistently across the cohort.</li>
<li><strong>Selection Criteria:</strong> Hits are selected based on a
custom <strong>label_score</strong> (<span class="math inline">\(|logFC|
\times -log10(P)\)</span>) to prioritize proteins in the extreme
“corners” of the plot.</li>
</ul>
</div>
<div id="heatmap" class="section level3">
<h3>Heatmap</h3>
<pre class="r fold-hide"><code># 1. Get the names of the top 30 proteins by significance
top_hits &lt;- de_results %&gt;% 
  slice_min(order_by = adj.P.Val, n = 20) %&gt;% 
  pull(Protein)

# 2. Subset your original data
# Assumes Olink_data has proteins as rows and sample IDs as columns
plot_matrix &lt;- Olink_data[top_hits, ]

# 3. Create annotation for the columns (samples)
annotation_col &lt;- data.frame(Status = Clinical_data$Group)
rownames(annotation_col) &lt;- colnames(Olink_data)

# 4. Plot
pheatmap(plot_matrix, 
         scale = &quot;row&quot;,                      # Normalize by protein to see relative changes
         annotation_col = annotation_col, 
         show_colnames = FALSE,
         main = &quot;Top 20 Proteins: SSc-PAH vs SSc-noPAH&quot;,
         color = my_palette)</code></pre>
<p><img src="figure/Olink-GB.Rmd/heatmap-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-heatmap-1">
Past versions of heatmap-1.png
</button>
</p>
<div id="fig-heatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/cb62f2a4767ff709d6940b5741d89ebe69302574/docs/figure/Olink-GB.Rmd/heatmap-1.png" target="_blank">cb62f2a</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/558ae77c852b60771a5ce16de896a786bab48a63/docs/figure/Olink-GB.Rmd/heatmap-1.png" target="_blank">558ae77</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<blockquote>
<p><strong>Key Finding:</strong> Clustering analysis reveals distinct
protein expression “blocks” that characterize the SSc-PAH group,
highlighting coordinated molecular shifts between the two
conditions.</p>
</blockquote>
<ul>
<li><strong>How to interpret this plot:</strong> Rows are proteins and
columns are samples. Red indicates higher expression relative to the row
average, and blue indicates lower expression. The trees (dendrograms)
group similar samples and similar proteins together.</li>
<li><strong>A Heatmap can show you:</strong> Protein “signatures.” It
reveals if sub-groups of patients exist within your clinical groups and
identifies which proteins move in unison.</li>
<li><strong>Selection Criteria:</strong> The top 20 proteins with the
lowest Adjusted P-values (FDR) were selected for visualization.</li>
</ul>
</div>
<div id="boxplot" class="section level3">
<h3>Boxplot</h3>
<pre class="r fold-hide"><code># --- STEP 0: Selection Logic ---
top_pval &lt;- de_results %&gt;% slice_min(order_by = adj.P.Val, n = 5) %&gt;% pull(Protein)
top_fc   &lt;- de_results %&gt;% slice_max(order_by = abs(logFC), n = 5) %&gt;% pull(Protein)
top_score &lt;- de_results %&gt;% slice_max(order_by = label_score, n = 5) %&gt;% pull(Protein)
protein_list &lt;- unique(c(top_pval, top_fc, top_score))

# --- STEP 1: Create plot_data_long from scratch ---
# This ensures the object exists in the local environment
plot_data_long &lt;- Olink_data[protein_list, ] %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Protein&quot;) %&gt;%
  # Convert to long format (Barcode matches your Clinical_data column)
  pivot_longer(-Protein, names_to = &quot;Barcode&quot;, values_to = &quot;NPX&quot;) %&gt;%
  # Merge with clinical metadata
  left_join(Clinical_data, by = &quot;Barcode&quot;)

# --- STEP 2: Create the multi-line labels ---
plot_labels &lt;- de_results %&gt;%
  filter(Protein %in% protein_list) %&gt;%
  mutate(Facet_Label = paste0(
    Protein, 
    &quot;\nadj.P: &quot;, formatC(adj.P.Val, format = &quot;e&quot;, digits = 2),
    &quot;\nlog2FC: &quot;, round(logFC, 2),
    &quot;\nScore: &quot;, round(label_score, 2)
  )) %&gt;%
  select(Protein, Facet_Label)

# --- STEP 3: Add labels to the data ---
plot_data_long &lt;- plot_data_long %&gt;%
  left_join(plot_labels, by = &quot;Protein&quot;)

# --- STEP 4: Generate the Grid ---
ggplot(plot_data_long, aes(x = Group, y = NPX, fill = Group)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.7, color = &quot;black&quot;, size = 0.3) +
  geom_jitter(width = 0.2, size = 1, alpha = 0.4, color = &quot;grey20&quot;) +
  facet_wrap(~Facet_Label, scales = &quot;free_y&quot;, ncol = 4) + 
  theme_bw() +
  scale_fill_manual(values = c(&quot;SSc-PAH&quot; = &quot;firebrick3&quot;, &quot;SSc-noPAH&quot; = &quot;dodgerblue3&quot;)) +
  theme(
    legend.position = &quot;none&quot;,
    strip.text = element_text(size = 8, face = &quot;bold&quot;, lineheight = 1.1),
    strip.background = element_rect(fill = &quot;grey95&quot;),
    axis.text.x = element_text(angle = 45, hjust = 1, size = 9),
    panel.grid.minor = element_blank()
  ) +
  labs(
    title = &quot;Key Differential Proteins: SSc-PAH vs SSc-noPAH&quot;,
    subtitle = &quot;Metrics: Adjusted P-value, Log2 Fold Change, and Combined Score&quot;,
    x = NULL, 
    y = &quot;Normalized Protein Expression (NPX)&quot;
  )</code></pre>
<p><img src="figure/Olink-GB.Rmd/boxplot-1.png" width="3600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-boxplot-1">
Past versions of boxplot-1.png
</button>
</p>
<div id="fig-boxplot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/9d5e0171c682bbe751613e473e6f589961538be4/docs/figure/Olink-GB.Rmd/boxplot-1.png" target="_blank">9d5e017</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/1e15a4ce30f6a3855a6560624fd3bdb447dc1eb2/docs/figure/Olink-GB.Rmd/boxplot-1.png" target="_blank">1e15a4c</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/558ae77c852b60771a5ce16de896a786bab48a63/docs/figure/Olink-GB.Rmd/boxplot-1.png" target="_blank">558ae77</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<blockquote>
<p><strong>Key Finding:</strong> Individual protein distributions
confirm that the top-ranked hits provide a consistent separation between
SSc-PAH and SSc-noPAH, with minimal influence from extreme outliers.</p>
</blockquote>
<ul>
<li><strong>How to interpret this plot:</strong> These show the exact
NPX (Normalized Protein Expression) values for specific proteins. The
box represents the interquartile range, and the horizontal line is the
median.</li>
<li><strong>A Boxplot can show you:</strong> The “cleanliness” of a
discovery. It allows you to see if the group difference is a consistent
shift across all patients or driven by only a few individuals.</li>
<li><strong>Selection Criteria:</strong> A combination of the top 5
proteins by P-value, Fold Change, and combined Score was used to provide
a balanced overview of the data.</li>
</ul>
</div>
<div id="save-top-labels" class="section level3">
<h3>Save top labels</h3>
<pre class="r"><code># Save the list of top hits used for labeling and boxplots
saveRDS(protein_list, file = file.path(output_dir_data, &quot;top_hits_DEA_proteins_list.rds&quot;))</code></pre>
</div>
<div id="ma-plot" class="section level3">
<h3>MA Plot</h3>
<pre class="r fold-hide"><code># Average expression (A) vs Log Fold Change (M)
# Use &#39;AveExpr&#39; from limma results
# 1. Create a subset for labeling: 
# Significant proteins with Average Expression &lt; 0
low_expr_labels &lt;- de_results %&gt;%
  filter(AveExpr &lt; 0 &amp; Significance != &quot;Not Significant&quot;)

# 2. Plot
ggplot(de_results, aes(x = AveExpr, y = logFC, color = Significance)) +
  geom_point(alpha = 0.4, size = 1.5) +
  scale_color_manual(values = c(&quot;Down-regulated&quot; = &quot;navy&quot;, 
                                &quot;Not Significant&quot; = &quot;grey70&quot;, 
                                &quot;Up-regulated&quot; = &quot;firebrick3&quot;)) +
  
  # Add labels for low expression proteins
  geom_text_repel(data = low_expr_labels,
                  aes(label = Protein),
                  size = 3,
                  color = &quot;black&quot;,
                  max.overlaps = 20,
                  box.padding = 0.5,
                  segment.size = 0.2) +
  
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;) +
  # Add a vertical line at the 0 threshold for clarity
  geom_vline(xintercept = 0, linetype = &quot;dotted&quot;, color = &quot;grey50&quot;) +
  
  theme_minimal() +
  labs(title = &quot;MA Plot: Log Fold Change vs. Average Expression&quot;,
       subtitle = &quot;Labels indicate significant proteins with log2 NPX &lt; 0&quot;,
       x = &quot;Average Expression (log2 NPX)&quot;,
       y = &quot;Log2 Fold Change&quot;)</code></pre>
<p><img src="figure/Olink-GB.Rmd/MA_plot-1.png" width="672" style="display: block; margin: auto;" />
&gt; <strong>Key Finding:</strong> The distribution of significant hits
across the range of average expression values suggests that the findings
are not biased by protein abundance or low-signal technical noise. Data
normalization is not perfect.</p>
<ul>
<li><strong>How to interpret this plot:</strong> The x-axis is the
Average Expression (signal intensity), and the y-axis is the Log Fold
Change. A balanced “cloud” centered at zero on the y-axis indicates
successful data normalization.</li>
<li><strong>An MA Plot can show you:</strong> If your significant hits
are concentrated in low-abundance proteins (which may be noisier) or if
they are distributed across the entire detection range of the Olink
panel.</li>
<li><strong>Selection Criteria:</strong> All proteins are plotted;
significant proteins with lower average expression are labeled to
monitor potential signal-to-noise issues.</li>
</ul>
<p>!! Doesn’t make much sense with NPX sine it is relative and not
absolute.</p>
</div>
</div>
<div id="protein-protein-interaction" class="section level2">
<h2>Protein Protein Interaction</h2>
<pre class="r hide-fold"><code># 1. Get your top 30 proteins
top_proteins &lt;- de_results %&gt;%
  slice_min(order_by = adj.P.Val, n = 30) %&gt;%
  pull(Protein) %&gt;%
  as.character()

# 2. Run the function
dummy_output &lt;- rba_string_network_image(
  ids = top_proteins, 
  species = 9606, 
  image_format = &quot;highres_image&quot;
)

# 3. Locate the downloaded file
# workflowr runs from the project root, so the rbioapi folder is created there
source_path &lt;- here::here(&quot;rbioapi&quot;, &quot;string_network_image.png&quot;)

# 4. Define the target destination
assets_dir &lt;- here::here(&quot;docs&quot;, &quot;assets&quot;)        
if (!dir.exists(assets_dir)) dir.create(assets_dir)       
target_path &lt;- file.path(assets_dir, &quot;ppi_network.png&quot;)

# 5. Move the file and clean up
if (file.exists(source_path)) {
  file.copy(source_path, target_path, overwrite = TRUE)
  unlink(here::here(&quot;rbioapi&quot;), recursive = TRUE) # Delete the auto-generated folder
}</code></pre>
<p><img src="assets/ppi_network.png" width="100%" alt="STRING Protein-Protein Interaction Network"></p>
<blockquote>
<p><strong>Key Finding:</strong> While some hits form clusters with
eachother, many hits are singletons. This suggests that many biological
processes are involved and not just one hub.</p>
</blockquote>
<ul>
<li><strong>How to interpret this plot:</strong> Circles (nodes) are
proteins, and lines (edges) are known interactions. Thicker lines
indicate higher confidence based on experimental evidence and
databases.</li>
<li><strong>A PPI Network can show you:</strong> If your hits are
biologically related. “Hub” proteins (those with many connections) often
represent the most central regulators of the disease process.</li>
<li><strong>Selection Criteria:</strong> The network is generated from
the top 30 proteins ranked by Adjusted P-value.</li>
</ul>
</div>
<div id="enrichment-analysis" class="section level2">
<h2>Enrichment Analysis</h2>
<div id="fetch-string-enrichment-results" class="section level3">
<h3>Fetch STRING Enrichment Results</h3>
<pre class="r"><code># 1. We use the same top 30 proteins used for the PPI network
top_proteins_enrich &lt;- de_results %&gt;%
  slice_min(order_by = adj.P.Val, n = 30) %&gt;%
  pull(Protein) %&gt;%
  as.character()

# 2. Call the STRING API to get functional enrichment
# This creates the &#39;enrichment_results&#39; object that was missing
enrichment_results &lt;- rba_string_enrichment(
  ids = top_proteins_enrich,
  #background = de_results$Protein, #background defines to what the expression is compared and thus when it is considered enriched or depleted
  species = 9606
)</code></pre>
<pre class="r fold-hide"><code># 1. Combine the list of results into a single data frame
# rba_string_enrichment returns a list of tables; bind_rows stacks them
enrichment_all &lt;- enrichment_results %&gt;% bind_rows()

# 2. Filter for specific biological categories
# We use categories confirmed to exist in your data: Function, Component, and Keyword
enrichment_clean &lt;- enrichment_all %&gt;%
  filter(category %in% c(&quot;Function&quot;, &quot;Component&quot;, &quot;Keyword&quot;)) %&gt;%
  arrange(fdr) %&gt;% # Sort by significance (lowest False Discovery Rate first)
  
  # Safely select columns: any_of() prevents errors if column names vary across versions
  # It looks for &#39;count&#39;, &#39;number_of_nodes&#39;, or &#39;number_of_genes&#39;
  select(term, description, fdr, category, 
         any_of(c(&quot;count&quot;, &quot;number_of_nodes&quot;, &quot;number_of_genes&quot;))) %&gt;%
  
  # Rename the 5th column to &#39;Count&#39; so our plotting code always works
  rename(Count = 5)

# 3. Prepare data for plotting
# We take the top 15 most significant pathways and fix mathematical issues
plot_data &lt;- head(enrichment_clean, 15) %&gt;%
  mutate(
    fdr = as.numeric(fdr),
    # If FDR is 0, replace with a tiny number to allow -log10 calculation (which fails at 0)
    fdr = ifelse(fdr &lt;= 0, 1e-10, fdr), 
    # Calculate the negative log10 of the FDR for better visualization (higher = more significant)
    neg_log10_fdr = -log10(fdr)
  )

# 4. Create the Lollipop Plot
# This is a cleaner alternative to a bar chart for categorical data
ggplot(plot_data, 
       aes(x = reorder(description, neg_log10_fdr), y = neg_log10_fdr)) +
  
  # Draw the &quot;stick&quot; of the lollipop
  geom_segment(aes(xend = reorder(description, neg_log10_fdr), yend = 0), 
               color = &quot;grey50&quot;) +
  
  # Draw the &quot;candy&quot; (point) of the lollipop
  # Size of point represents number of proteins; color represents the database source
  geom_point(aes(size = Count, color = category), alpha = 0.8) +
  
  # Flip coordinates to make long pathway descriptions easy to read
  coord_flip() +
  
  # Use a clean, professional theme for the report
  theme_minimal() +
  
  # Apply a distinct color palette for the categories
  scale_color_brewer(palette = &quot;Dark2&quot;) + 
  
  # Add descriptive labels for the report
  labs(
    title = &quot;Functional Enrichment: SSc-PAH vs SSc-noPAH&quot;,
    subtitle = &quot;Top 15 Molecular Functions &amp; Components&quot;,
    x = NULL, # Description is self-explanatory on the Y-axis
    y = &quot;-log10(FDR)&quot;,
    size = &quot;Protein Count&quot;,
    color = &quot;Category&quot;
  )</code></pre>
<p><img src="figure/Olink-GB.Rmd/Functional_Enrichment-1.png" width="672" style="display: block; margin: auto;" />
&gt; <strong>Key Finding:</strong> Significant enrichment in specific
molecular functions and cellular components suggests that the proteomic
changes in SSc-PAH are concentrated in these biological processes.</p>
<ul>
<li><strong>How to interpret this plot:</strong> The length of the line
(x-axis) indicates the significance (-log10 FDR). The dot size
represents the number of proteins involved in that specific term.</li>
<li><strong>A Lollipop Plot can show you:</strong> The “big picture”
biology. It groups individual proteins into functional categories,
making it easier to see which systems (e.g., “Extracellular Matrix” or
“Cytokine Activity”) are most affected.</li>
<li><strong>Selection Criteria:</strong> Displays the top 15 most
significant terms across the Function, Component, and Keyword categories
from the STRING database.</li>
</ul>
</div>
<div id="table-of-top-15-enrichment-groups-and-proteins"
class="section level3">
<h3>Table of Top 15 Enrichment Groups and Proteins</h3>
</div>
<div id="generate-a-clean-table-of-top-15-enrichment-groups"
class="section level3">
<h3>Generate a Clean Table of Top 15 Enrichment Groups</h3>
<pre class="r fold-hide"><code># 1. Take the top 15 groups
top_15_groups &lt;- head(enrichment_clean, 15)

# 2. Join with raw data and format the protein list
enrichment_table_clean &lt;- top_15_groups %&gt;%
  left_join(
    enrichment_all %&gt;% select(description, inputGenes), 
    by = &quot;description&quot;
  ) %&gt;%
  mutate(
    # Remove c(), quotes, and extra spaces
    # This turns c(&quot;ACE2&quot;, &quot;HGF&quot;) into &quot;ACE2, HGF&quot;
    Proteins = gsub(&#39;c\\(|\\)|&quot;|\\s&#39;, &#39;&#39;, inputGenes),
    # Optional: Replace commas with &quot;, &quot; for better line-wrapping in tables
    Proteins = gsub(&#39;,&#39;, &#39;, &#39;, Proteins),
    # Format FDR to a clean scientific notation
    FDR = formatC(as.numeric(fdr), format = &quot;e&quot;, digits = 2)
  ) %&gt;%
  # 3. Select and rename for the final display
  select(
    Category = category,
    Description = description,
    Count = Count,
    FDR,
    `Associated Proteins` = Proteins
  )

# 4. Display with kableExtra for a high-quality look
knitr::kable(enrichment_table_clean, 
             caption = &quot;Table 1: Top 15 Functional Enrichment Terms&quot;,
             booktabs = TRUE) %&gt;%
  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;), 
                full_width = F, 
                position = &quot;left&quot;) %&gt;%
  column_spec(5, width = &quot;30em&quot;) # This prevents the protein list from making the table too wide</code></pre>
<table class="table table-striped table-hover table-condensed" style="width: auto !important; ">
<caption>
Table 1: Top 15 Functional Enrichment Terms
</caption>
<thead>
<tr>
<th style="text-align:left;">
Category
</th>
<th style="text-align:left;">
Description
</th>
<th style="text-align:right;">
Count
</th>
<th style="text-align:left;">
FDR
</th>
<th style="text-align:left;">
Associated Proteins
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Keyword
</td>
<td style="text-align:left;">
Signal
</td>
<td style="text-align:right;">
24
</td>
<td style="text-align:left;">
3.04e-11
</td>
<td style="text-align:left;width: 30em; ">
DCN, LAG3, EBI3, HGF, RAMP2, CD302, CFC1, JTB, TNFRSF10B, SNED1, LRRN1,
PRELP, GFRA1, RSPO1, RSPO3, EPHA2, RAET1L, SFTPA2, LAYN, TNFRSF19,
TAFA5, ACE2, EFNA4, ADM
</td>
</tr>
<tr>
<td style="text-align:left;">
Keyword
</td>
<td style="text-align:left;">
Disulfide bond
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:left;">
3.90e-10
</td>
<td style="text-align:left;width: 30em; ">
DCN, LAG3, HGF, RAMP2, CD302, CFC1, SCARB2, TNFRSF10B, SNED1, LRRN1,
TNFSF13, PRELP, GFRA1, RSPO1, RSPO3, EPHA2, RAET1L, LAYN, TNFRSF19,
EDA2R, ACE2, EFNA4, ADM
</td>
</tr>
<tr>
<td style="text-align:left;">
Keyword
</td>
<td style="text-align:left;">
Glycoprotein
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:left;">
8.79e-08
</td>
<td style="text-align:left;width: 30em; ">
DCN, LAG3, EBI3, HGF, RAMP2, CD302, CFC1, SCARB2, SNED1, LRRN1, TNFSF13,
PRELP, GFRA1, RSPO1, RSPO3, EPHA2, RAET1L, LAYN, TNFRSF19, EDA2R, TAFA5,
ACE2, EFNA4
</td>
</tr>
<tr>
<td style="text-align:left;">
Component
</td>
<td style="text-align:left;">
Cell surface
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:left;">
8.93e-05
</td>
<td style="text-align:left;width: 30em; ">
LAG3, EBI3, RAMP2, CD302, CFC1, TNFRSF10B, GFRA1, EPHA2, RAET1L, LAYN,
ACE2
</td>
</tr>
<tr>
<td style="text-align:left;">
Component
</td>
<td style="text-align:left;">
Extracellular region
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:left;">
1.10e-04
</td>
<td style="text-align:left;width: 30em; ">
DCN, LAG3, EBI3, HGF, SCARB2, SNED1, PRELP, GFRA1, RSPO1, RSPO3, SFTPA2,
TAFA5, ACE2, ADM
</td>
</tr>
<tr>
<td style="text-align:left;">
Component
</td>
<td style="text-align:left;">
Extracellular region
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:left;">
1.10e-04
</td>
<td style="text-align:left;width: 30em; ">
DCN, LAG3, EBI3, HGF, CFC1, SCARB2, SNED1, LRRN1, TNFSF13, PRELP, GFRA1,
RSPO1, RSPO3, RAET1L, SFTPA2, LGALS7, TAFA5, ACE2, EFNA4, ADM
</td>
</tr>
<tr>
<td style="text-align:left;">
Keyword
</td>
<td style="text-align:left;">
Secreted
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:left;">
1.80e-04
</td>
<td style="text-align:left;width: 30em; ">
DCN, LAG3, EBI3, CFC1, SNED1, TNFSF13, PRELP, RSPO1, RSPO3, TAFA5, ACE2,
EFNA4, ADM
</td>
</tr>
<tr>
<td style="text-align:left;">
Function
</td>
<td style="text-align:left;">
Death receptor activity
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
4.50e-03
</td>
<td style="text-align:left;width: 30em; ">
TNFRSF10B, TNFRSF19, EDA2R
</td>
</tr>
<tr>
<td style="text-align:left;">
Function
</td>
<td style="text-align:left;">
Signaling receptor binding
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:left;">
4.50e-03
</td>
<td style="text-align:left;width: 30em; ">
LAG3, EBI3, HGF, CFC1, SNED1, TNFSF13, GFRA1, RSPO1, RSPO3, TAFA5,
EFNA4, ADM
</td>
</tr>
<tr>
<td style="text-align:left;">
Keyword
</td>
<td style="text-align:left;">
GPI-anchor
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
8.20e-03
</td>
<td style="text-align:left;width: 30em; ">
CFC1, GFRA1, RAET1L, EFNA4
</td>
</tr>
<tr>
<td style="text-align:left;">
Function
</td>
<td style="text-align:left;">
Signaling receptor activity
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:left;">
1.10e-02
</td>
<td style="text-align:left;width: 30em; ">
LAG3, EBI3, RAMP2, CD302, SCARB2, TNFRSF10B, GFRA1, EPHA2, TNFRSF19,
EDA2R, EFNA4
</td>
</tr>
<tr>
<td style="text-align:left;">
Function
</td>
<td style="text-align:left;">
Transmembrane signaling receptor activity
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:left;">
1.12e-02
</td>
<td style="text-align:left;width: 30em; ">
LAG3, EBI3, RAMP2, SCARB2, TNFRSF10B, GFRA1, EPHA2, TNFRSF19, EDA2R,
EFNA4
</td>
</tr>
<tr>
<td style="text-align:left;">
Keyword
</td>
<td style="text-align:left;">
Receptor
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
1.89e-02
</td>
<td style="text-align:left;width: 30em; ">
RAMP2, CD302, SCARB2, TNFRSF10B, GFRA1, EPHA2, TNFRSF19, EDA2R, ACE2
</td>
</tr>
<tr>
<td style="text-align:left;">
Keyword
</td>
<td style="text-align:left;">
Host cell receptor for virus entry
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:left;">
1.89e-02
</td>
<td style="text-align:left;width: 30em; ">
SCARB2, EPHA2, ACE2
</td>
</tr>
<tr>
<td style="text-align:left;">
Function
</td>
<td style="text-align:left;">
Glycosaminoglycan binding
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:left;">
2.84e-02
</td>
<td style="text-align:left;width: 30em; ">
DCN, PRELP, RSPO1, RSPO3, LAYN
</td>
</tr>
<tr>
<td style="text-align:left;">
Component
</td>
<td style="text-align:left;">
Cell periphery
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:left;">
3.39e-02
</td>
<td style="text-align:left;width: 30em; ">
DCN, LAG3, EBI3, RAMP2, CD302, CFC1, SCARB2, JTB, TNFRSF10B, SNED1,
LRRN1, PRELP, GFRA1, EPHA2, RAET1L, PALM2, TNFRSF19, EDA2R, ACE2, EFNA4
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="gene-set-enrichment-analysis-from-the-dea"
class="section level2">
<h2>Gene Set Enrichment Analysis from the DEA</h2>
<div id="prepare-ranked-list-for-gsea" class="section level3">
<h3>Prepare Ranked List for GSEA</h3>
<p>!! Reconsider selection based on t, adjp or fc</p>
<pre class="r fold-hide"><code># We need a clean, sorted, named vector with no NAs
gsea_input &lt;- de_results %&gt;%
  # Remove rows where Protein name or t-statistic is missing
  filter(!is.na(Protein), !is.na(t)) %&gt;%
  # If there are duplicate protein names, take the one with the highest absolute t-stat
  group_by(Protein) %&gt;%
  slice_max(order_by = abs(t), n = 1, with_ties = FALSE) %&gt;%
  ungroup() %&gt;%
  # Sort descending by t-statistic
  arrange(desc(t))

# Create the named vector
ranked_vec &lt;- gsea_input$t
names(ranked_vec) &lt;- gsea_input$Protein

# FINAL CHECK: Remove any remaining NAs or non-finite values (like Inf)
ranked_vec &lt;- ranked_vec[is.finite(ranked_vec)]</code></pre>
</div>
<div id="run-gsea" class="section level3">
<h3>Run GSEA</h3>
<p>!! Consider using another subcategory than BP</p>
<pre class="r fold-hide"><code>### 1. Get GO Biological Process Gene Sets
go_genesets &lt;- msigdbr(species = &quot;Homo sapiens&quot;, category = &quot;C5&quot;, subcategory = &quot;BP&quot;) %&gt;%
  select(gs_name, gene_symbol)

### 2. Run GSEA with GO-BP
gsea_res &lt;- GSEA(
  geneList = ranked_vec,
  TERM2GENE = go_genesets,
  pvalueCutoff = 1,      # See everything first
  minGSSize = 5,         # Suitable for Olink
  maxGSSize = 500,
  eps = 0,               # Better precision for p-values
  nPermSimple = 10000,   # More stable permutations
  verbose = FALSE
)

# Check the results
gsea_results_df &lt;- as.data.frame(gsea_res)
# print(paste(&quot;Number of GO terms found:&quot;, nrow(gsea_results_df)))</code></pre>
</div>
<div id="filter-gsea-results" class="section level3">
<h3>Filter GSEA Results</h3>
<p>This filter is too strict. Instead for searching for significant
differences lets’s search for trends and label them as such. remplace
FDR with uncorrected p-value !! Rather use adjusted p-value &lt; 0.2</p>
<pre class="r"><code># 1. Choose your filter: Trends (p-value) vs. Significant (p.adjust)
gsea_trends &lt;- gsea_results_df %&gt;%
  # Filter for trends (raw p-value &lt; 0.05)
  filter(pvalue &lt; 0.05) %&gt;% 
  
  # --- UNCOMMENT BELOW TO SWITCH BACK TO STRICT SIGNIFICANCE ---
  # filter(p.adjust &lt; 0.05) %&gt;% 
  # --------------------------------------------------------------
  
  # 2. Separate the NES into &#39;Activated&#39; and &#39;Suppressed&#39;
  mutate(Status = ifelse(NES &gt; 0, &quot;Up-regulated in SSc-PAH&quot;, &quot;Down-regulated in SSc-PAH&quot;)) %&gt;%
  
  # 3. Clean the GO Term names (Remove GOBP_ and underscores)
  mutate(Description = gsub(&quot;GOBP_&quot;, &quot;&quot;, Description),
         Description = gsub(&quot;_&quot;, &quot; &quot;, Description)) %&gt;%
  
  # 4. Order by effect size (NES)
  arrange(desc(abs(NES)))

# Check how many trending terms remain
print(paste(&quot;Trending GO terms (raw p &lt; 0.05) found:&quot;, nrow(gsea_trends)))</code></pre>
<pre><code>[1] &quot;Trending GO terms (raw p &lt; 0.05) found: 261&quot;</code></pre>
<pre class="r"><code># Save the trending results for your supplement
write.csv(gsea_trends, file = file.path(output_dir_data, &quot;GSEA_GO_BP_trends.csv&quot;), row.names = FALSE)</code></pre>
</div>
<div id="visualization-of-gsea" class="section level3">
<h3>Visualization of GSEA</h3>
<pre class="r"><code># 1. Prepare the data with wrapped labels
top_10_trends_wrapped &lt;- gsea_trends %&gt;%
  group_by(Status) %&gt;%
  slice_max(order_by = abs(NES), n = 10) %&gt;%
  ungroup() %&gt;%
  mutate(
    # Wrap strings at 40 characters
    # You can change 40 to 30 if you want them even narrower
    Description_Wrapped = str_wrap(Description, width = 40)
  )

# 2. Create the plot
ggplot(top_10_trends_wrapped, aes(x = NES, y = reorder(Description_Wrapped, NES))) +
  geom_point(aes(size = setSize, color = pvalue)) +
  
  # Standard color scale: red for significant/high-p, blue for low-p (or vice versa)
  scale_color_gradient(low = &quot;indianred3&quot;, high = &quot;navyblue&quot;) +
  
  # Add a vertical line at 0
  geom_vline(xintercept = 0, linetype = &quot;dashed&quot;, color = &quot;grey50&quot;) +
  
  theme_minimal() +
  theme(
    # Increase line spacing for the wrapped y-axis labels
    axis.text.y = element_text(lineheight = 0.8, size = 9),
    panel.grid.minor = element_blank()
  ) +
  labs(
    title = &quot;Top Biological Trends in SSc-PAH (GSEA)&quot;,
    subtitle = &quot;Top 10 Up/Down pathways (raw p-value &lt; 0.05)&quot;,
    x = &quot;Normalized Enrichment Score (NES)&quot;,
    y = NULL,
    size = &quot;Protein Count&quot;,
    color = &quot;P-value&quot;
  )</code></pre>
<p><img src="figure/Olink-GB.Rmd/Visualization_GSEA-1.png" width="768" style="display: block; margin: auto;" />
&gt; <strong>Key Finding:</strong> Several GO Biological Processes show
strong enrichment trends, indicating that SSc-PAH involves broad,
pathway-level shifts even in cases where individual proteins may not
reach strict significance.</p>
<ul>
<li><strong>How to interpret this plot:</strong> The x-axis is the NES
(Normalized Enrichment Score). Color indicates the p-value</li>
<li><strong>A GSEA Dot Plot can show you:</strong> Coordinated trends.
It is more powerful than looking at single proteins because it
aggregates signal from many related proteins to identify broader
biological themes.</li>
<li><strong>Selection Criteria:</strong> Filtered for trends (raw
p-value &lt; 0.05) to capture a wide breadth of biological signal,
selecting the top 10 Up and top 10 Down terms by NES.</li>
</ul>
<p>!!Make an emaplot to group the terms</p>
</div>
<div id="ridge-plot-_of-top-gsea-trends" class="section level3">
<h3>Ridge plot _of Top GSEA Trends</h3>
<pre class="r"><code># 1. Get the names of the pathways you used in your &quot;Trends&quot; plot
# (Assuming you want the top 15 by absolute NES)
target_pathways &lt;- gsea_trends %&gt;%
  arrange(desc(abs(NES))) %&gt;%
  slice_head(n = 20) %&gt;%
  pull(ID) # We use ID because clusterProfiler functions prefer the original GOBP_ ID

# 2. Force the ridgeplot to show ONLY these pathways
ridgeplot(gsea_res, showCategory = target_pathways) +
  theme_minimal() +
  geom_vline(xintercept = 0, linetype = &quot;dashed&quot;, color = &quot;red&quot;) +
  labs(title = &quot;Distribution of Trends in SSc-PAH&quot;,
       x = &quot;Differential Expression Score (t-statistic)&quot;,
       y = NULL)</code></pre>
<p><img src="figure/Olink-GB.Rmd/ridgeplotGSEA-1.png" width="768" style="display: block; margin: auto;" />
&gt; <strong>Key Finding:</strong> The observed biological trends show a
coordinated shift in protein expression across specific pathways,
suggesting that SSc-PAH is driven by broad functional changes rather
than isolated protein fluctuations.</p>
<ul>
<li><strong>How to interpret this plot:</strong> Each “ridge” represents
a GO pathway. The x-axis indicates the t-statistic (Differential
Expression Score). A peak shifted to the right of zero indicates
enrichment in SSc-PAH, while a peak to the left indicates
depletion.</li>
<li><strong>A Ridge Plot can show you:</strong> The “coordination” of a
biological process. A narrow, tall peak indicates that all proteins in
that pathway move together consistently, whereas a wide peak indicates
high variance within that specific biological function.</li>
<li><strong>Selection Criteria:</strong> The top 20 pathways are
displayed based on the highest absolute Normalized Enrichment Score
(NES) from the trending GSEA results (<span class="math inline">\(p &lt;
0.05\)</span>).</li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.2 (2025-10-31)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 24.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] kableExtra_1.4.0       patchwork_1.3.2        pheatmap_1.0.13       
 [4] EnhancedVolcano_1.28.2 ggrepel_0.9.6          ggridges_0.5.7        
 [7] enrichplot_1.30.4      clusterProfiler_4.18.4 msigdbr_25.1.1        
[10] rbioapi_0.8.3          factoextra_1.0.7       FactoMineR_2.13       
[13] limma_3.66.0           readxl_1.4.5           RColorBrewer_1.1-3    
[16] magrittr_2.0.4         here_1.0.2             lubridate_1.9.4       
[19] forcats_1.0.1          stringr_1.5.2          dplyr_1.1.4           
[22] purrr_1.1.0            readr_2.1.5            tidyr_1.3.1           
[25] tibble_3.3.0           ggplot2_4.0.0          tidyverse_2.0.0       
[28] workflowr_1.7.2       

loaded via a namespace (and not attached):
  [1] splines_4.5.2           later_1.4.4             ggplotify_0.1.3        
  [4] R.oo_1.27.1             cellranger_1.1.0        polyclip_1.10-7        
  [7] lifecycle_1.0.4         rstatix_0.7.3           rprojroot_2.1.1        
 [10] processx_3.8.6          lattice_0.22-7          MASS_7.3-65            
 [13] flashClust_1.01-2       backports_1.5.0         sass_0.4.10            
 [16] rmarkdown_2.30          jquerylib_0.1.4         yaml_2.3.10            
 [19] httpuv_1.6.16           otel_0.2.0              ggtangle_0.1.1         
 [22] cowplot_1.2.0           DBI_1.2.3               abind_1.4-8            
 [25] multcomp_1.4-29         R.utils_2.13.0          BiocGenerics_0.56.0    
 [28] yulab.utils_0.2.4       TH.data_1.1-4           tweenr_2.0.3           
 [31] rappdirs_0.3.3          sandwich_3.1-1          git2r_0.36.2           
 [34] gdtools_0.5.0           IRanges_2.44.0          S4Vectors_0.48.0       
 [37] tidytree_0.4.7          svglite_2.2.2           codetools_0.2-20       
 [40] xml2_1.4.1              DOSE_4.4.0              DT_0.34.0              
 [43] ggforce_0.5.0           tidyselect_1.2.1        aplot_0.2.9            
 [46] farver_2.1.2            stats4_4.5.2            Seqinfo_1.0.0          
 [49] jsonlite_2.0.0          Formula_1.2-5           survival_3.8-3         
 [52] emmeans_2.0.1           systemfonts_1.3.1       tools_4.5.2            
 [55] ggnewscale_0.5.2        treeio_1.34.0           Rcpp_1.1.0             
 [58] glue_1.8.0              xfun_0.53               qvalue_2.42.0          
 [61] withr_3.0.2             fastmap_1.2.0           callr_3.7.6            
 [64] digest_0.6.37           timechange_0.3.0        R6_2.6.1               
 [67] gridGraphics_0.5-1      estimability_1.5.1      textshaping_1.0.4      
 [70] GO.db_3.22.0            RSQLite_2.4.3           R.methodsS3_1.8.2      
 [73] generics_0.1.4          fontLiberation_0.1.0    data.table_1.17.8      
 [76] httr_1.4.7              htmlwidgets_1.6.4       scatterpie_0.2.6       
 [79] scatterplot3d_0.3-44    whisker_0.4.1           pkgconfig_2.0.3        
 [82] gtable_0.3.6            blob_1.2.4              S7_0.2.0               
 [85] XVector_0.50.0          htmltools_0.5.8.1       carData_3.0-6          
 [88] fontBitstreamVera_0.1.1 fgsea_1.36.2            multcompView_0.1-11    
 [91] scales_1.4.0            Biobase_2.70.0          leaps_3.2              
 [94] png_0.1-8               ggfun_0.2.0             knitr_1.50             
 [97] rstudioapi_0.17.1       tzdb_0.5.0              reshape2_1.4.5         
[100] coda_0.19-4.1           nlme_3.1-168            curl_7.0.0             
[103] cachem_1.1.0            zoo_1.8-14              parallel_4.5.2         
[106] AnnotationDbi_1.72.0    pillar_1.11.1           grid_4.5.2             
[109] vctrs_0.6.5             ggpubr_0.6.2            promises_1.4.0         
[112] car_3.1-5               tidydr_0.0.6            xtable_1.8-4           
[115] cluster_2.1.8.1         evaluate_1.0.5          mvtnorm_1.3-3          
[118] cli_3.6.5               compiler_4.5.2          rlang_1.1.6            
[121] crayon_1.5.3            ggsignif_0.6.4          labeling_0.4.3         
[124] ps_1.9.1                getPass_0.2-4           plyr_1.8.9             
[127] fs_1.6.6                ggiraph_0.9.5           stringi_1.8.7          
[130] viridisLite_0.4.2       BiocParallel_1.44.0     assertthat_0.2.1       
[133] babelgene_22.9          Biostrings_2.78.0       lazyeval_0.2.2         
[136] GOSemSim_2.36.0         fontquiver_0.2.1        Matrix_1.7-4           
[139] hms_1.1.4               bit64_4.6.0-1           KEGGREST_1.50.0        
[142] statmod_1.5.1           broom_1.0.10            igraph_2.2.1           
[145] memoise_2.0.1           bslib_0.9.0             ggtree_4.0.4           
[148] fastmatch_1.1-8         bit_4.6.0               gson_0.1.0             
[151] ape_5.8-1              </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
