<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="alexander-gysin (alexander.gysin@usz.ch)" />


<title>Olink-GB</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">omics-training</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Olink-GB.html">Olink Analysis</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/alexander-gysin/omics-training">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Olink-GB</h1>
<h4 class="author">alexander-gysin (<a
href="mailto:alexander.gysin@usz.ch"
class="email">alexander.gysin@usz.ch</a>)</h4>
<h4 class="date">18 February, 2026</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-02-18
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>omics-training/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.2). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20260210code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20260210)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20260210code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20260210)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomalexandergysinomicstrainingtree54609d1dd9114cc8b465e272da80a0ad698802f6targetblank54609d1a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/alexander-gysin/omics-training/tree/54609d1dd9114cc8b465e272da80a0ad698802f6" target="_blank">54609d1</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomalexandergysinomicstrainingtree54609d1dd9114cc8b465e272da80a0ad698802f6targetblank54609d1a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/alexander-gysin/omics-training/tree/54609d1dd9114cc8b465e272da80a0ad698802f6" target="_blank">54609d1</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/Olink-GB/

Untracked files:
    Untracked:  code/Olink-GB/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/_site.yml
    Deleted:    analysis/first-analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Olink-GB.Rmd</code>) and HTML
(<code>docs/Olink-GB.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/54609d1dd9114cc8b465e272da80a0ad698802f6/analysis/Olink-GB.Rmd" target="_blank">54609d1</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/9ba78e7f24bd98107a5e62a732537c23398a02c6/docs/Olink-GB.html" target="_blank">9ba78e7</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/48ac657607c15ead3162113aec9a8282d3f4072d/analysis/Olink-GB.Rmd" target="_blank">48ac657</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/cb4c6c9fa1e1879d68971ef4d5a5547b303dd84a/docs/Olink-GB.html" target="_blank">cb4c6c9</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/632763f2ea5f491d0af0fb371f7b885ce2effa59/analysis/Olink-GB.Rmd" target="_blank">632763f</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Hide all except certain code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/a45865e9f58830693dfd08e2de0229606879372e/docs/Olink-GB.html" target="_blank">a45865e</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/7ac5b7069b5bb3e8b41a4a9b9db7eff5c8e5973a/docs/Olink-GB.html" target="_blank">7ac5b70</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/8fd6e5e660effa56918a26a606bebc297a9f3acc/analysis/Olink-GB.Rmd" target="_blank">8fd6e5e</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/8635b5ce733aebf9d67222c435e28d98252cd87c/docs/Olink-GB.html" target="_blank">8635b5c</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/a5e69971d098f0f0523d3dbd6fd1ca87f406e5ab/analysis/Olink-GB.Rmd" target="_blank">a5e6997</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/8fc46d7030714fd488cb2247e34066b19133b4da/docs/Olink-GB.html" target="_blank">8fc46d7</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/c583496bc28a96d12873bad42a01ebfc03bb2490/analysis/Olink-GB.Rmd" target="_blank">c583496</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/a00cade57b936659b8f9850db0eeb521f55829c1/docs/Olink-GB.html" target="_blank">a00cade</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/608c7931d26de437762a9679acabd4d4c034e506/analysis/Olink-GB.Rmd" target="_blank">608c793</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexander-gysin/omics-training/7b8d3a56178290dcf33f811842a2ad3bb0144298/docs/Olink-GB.html" target="_blank">7b8d3a5</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/f2390c0024f04f88c8b8cc53cb2137884de9454e/analysis/Olink-GB.Rmd" target="_blank">f2390c0</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-17
</td>
<td>
wflow_publish("analysis/Olink-GB.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This serves as training to analyze Olink data</p>
</div>
<div id="setup" class="section level1">
<h1>Setup</h1>
<div id="knitr-configuration" class="section level3">
<h3>knitr configuration</h3>
</div>
<div id="directories" class="section level3">
<h3>Directories</h3>
<pre class="r"><code># Get current file name to make folder
current_file &lt;- &quot;Olink-GB&quot;
#Output paths
output_dir_data &lt;- here::here(&quot;output&quot;, current_file)
if (!dir.exists(output_dir_data)) dir.create(output_dir_data)

if (!dir.exists(here::here(&quot;docs&quot;, &quot;figure&quot;))) dir.create(here::here(&quot;docs&quot;, &quot;figure&quot;))

output_dir_figs &lt;- here::here(&quot;docs&quot;, &quot;figure&quot;, paste0(current_file, &quot;.Rmd&quot;))
if (!dir.exists(output_dir_figs)) dir.create(output_dir_figs)</code></pre>
</div>
<div id="packages" class="section level3">
<h3>Packages</h3>
<pre class="r"><code># Utility Packages
library(tidyverse)  # Contains dplyr (data manipulation), readr, tibble (modern df), ggplot2 and more
library(here)       # Uses the project root for file paths; stops &quot;File not found&quot; errors on other PCs
library(magrittr)   # Provides the pipe operator (%&gt;%) to make your code readable
library(RColorBrewer) # Provides professional, color-blind friendly palettes for your plots
library(readxl)     # Used to load excel files

# Differential Expression &amp; PCA Packages
library(limma)            # The gold standard for linear modeling and DEA (works for Olink/Proteomics)
library(stats)            # Built-in R package; provides prcomp() for PCA (no need to load explicitly)
library(FactoMineR)       # Multi Factor Analysis for multi-omics, calculates the contribution to the PCA
library(factoextra)       # PCA Visualization

# Visualization Packages
library(EnhancedVolcano)  # Specifically designed for high-quality, easy volcano plots
library(pheatmap)         # Quick and clean heatmaps to visualize clusters after PCA
library(patchwork)        # Essential for combining multiple plots (e.g., PCA next to a Volcano plot)</code></pre>
<p>#Data preparation &amp; QC ## Initialization ### Read data</p>
<pre class="r"><code>Olink_data &lt;- read.csv(here::here(&quot;data&quot;, &quot;quant.csv&quot;), sep = &quot;;&quot;)
Clinical_data &lt;- read_excel(here::here(&quot;data&quot;, &quot;Metadata_Olink_PL.xlsx&quot;))</code></pre>
</div>
<div id="define-color-palette" class="section level3">
<h3>Define color palette</h3>
<pre class="r"><code>my_palette &lt;- colorRampPalette(c(&quot;navy&quot;, &quot;whitesmoke&quot;, &quot;indianred&quot;))(100)</code></pre>
</div>
<div id="data-clean-up" class="section level2">
<h2>Data clean-up</h2>
<div id="formating-the-olink-file" class="section level3">
<h3>Formating the Olink file</h3>
<p>Protein names to rownames in the Olink file</p>
<pre class="r"><code>Olink_data &lt;- Olink_data %&gt;%
  column_to_rownames(&quot;SampleID&quot;) # Replace &quot;ProteinID&quot; with your actual column name</code></pre>
</div>
<div id="order-olink-data-according-to-metadata" class="section level3">
<h3>Order Olink data according to metadata</h3>
<pre class="r"><code>Olink_data &lt;- Olink_data[, Clinical_data$Barcode]

# Check if column names match metadata barcodes exactly
stopifnot(all(colnames(Olink_data) == Clinical_data$Barcode))</code></pre>
</div>
<div id="remove-the-olink_id-from-the-rownames" class="section level3">
<h3>Remove the Olink_ID from the rownames</h3>
<pre class="r"><code>rownames(Olink_data) &lt;- str_remove(rownames(Olink_data), &quot;_.*&quot;)</code></pre>
</div>
<div id="check-for-missing-values-in-the-olink-file"
class="section level3">
<h3>Check for missing values in the Olink file</h3>
<pre class="r"><code>sum(is.na(Olink_data))</code></pre>
<pre><code>[1] 0</code></pre>
<p>#Results ## PCA Analysis {.tabset}</p>
</div>
<div id="pca-analysis" class="section level3">
<h3>PCA Analysis</h3>
<pre class="r"><code># Note: PCA functions usually expect samples in ROWS, so we transpose with t()
pca_res &lt;- PCA(t(Olink_data), graph = FALSE)</code></pre>
</div>
</div>
<div id="pca-visualization" class="section level2">
<h2>PCA visualization</h2>
<div id="plotting-eigenvalues" class="section level3">
<h3>Plotting Eigenvalues</h3>
<pre class="r"><code>p_scree &lt;- fviz_eig(pca_res, addlabels = TRUE, barfill = &quot;navy&quot;, barcolor = &quot;navy&quot;)
p_scree</code></pre>
<p><img src="figure/Olink-GB.Rmd/eigenvalue%20vis-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-eigenvalue-vis-1">
Past versions of "eigenvalue vis-1.png"
</button>
</p>
<div id="fig-no-spaces-eigenvalue-vis-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/9ba78e7f24bd98107a5e62a732537c23398a02c6/docs/figure/Olink-GB.Rmd/eigenvalue vis-1.png" target="_blank">9ba78e7</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Ths plot shows the Eigenvalues of the individual Principal Component.
The higher the percentage the more variance is explained by the
Principal Component. PC1 and PC2 cumulatively explain 37.1% of the
variance, which is deemed acceptable.</p>
</div>
<div id="plotting-pca-by-group" class="section level3">
<h3>Plotting PCA by Group</h3>
<pre class="r"><code># 2. PCA Plot colored by Group
p_pca &lt;- fviz_pca_ind(pca_res,
             label = &quot;none&quot;, # Hide sample names for clarity
             habillage = as.factor(Clinical_data$Group), 
             palette = c(&quot;indianred&quot;, &quot;navy&quot;),
             addEllipses = TRUE, 
             invisible = &quot;quali&quot;) +
             theme_minimal() +
             labs(title = &quot;Proteomics PCA&quot;)
p_pca</code></pre>
<p><img src="figure/Olink-GB.Rmd/PCA%20Plot%20by%20group-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-PCA-Plot-by-group-1">
Past versions of "PCA Plot by group-1.png"
</button>
</p>
<div id="fig-no-spaces-PCA-Plot-by-group-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/9ba78e7f24bd98107a5e62a732537c23398a02c6/docs/figure/Olink-GB.Rmd/PCA Plot by group-1.png" target="_blank">9ba78e7</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>###Visualizing contribution by protein in a simple contribution
plot</p>
<pre class="r"><code># manually set a common limit
common_limit &lt;- 0.5
# Visualize the top 10 proteins contributing to PC1
p_top_proteins_PD1 &lt;- fviz_contrib(pca_res, choice = &quot;var&quot;, axes = 1, top = 10, fill = &quot;navy&quot;, color = &quot;navy&quot;, sort.val = &quot;asc&quot;) +
  theme_minimal() +
  labs(title = &quot;Top 10 Proteins Contributing to PC1&quot;,
       x = &quot;Proteins&quot;,
       y = &quot;Contribution [%]&quot;)+
  coord_flip(ylim = c(0, common_limit)) # Flips the plot and sets the scale

# Visualize the top 10 proteins contributing to PC2
p_top_proteins_PD2 &lt;- fviz_contrib(pca_res, choice = &quot;var&quot;, axes = 2, top = 10, fill = &quot;navy&quot;, color = &quot;navy&quot;, sort.val = &quot;asc&quot;) +
  theme_minimal() +
  labs(title = &quot;Top 10 Proteins Contributing to PC2&quot;,
       x = &quot;Proteins&quot;,
       y = &quot;Contribution [%]&quot;)+
  coord_flip(ylim = c(0, common_limit)) # Flips the plot and sets the scale

# / displays the plots in vertical order
p_top_proteins_PD1 / p_top_proteins_PD2</code></pre>
<p><img src="figure/Olink-GB.Rmd/top%2010%20proteins-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-top-10-proteins-1">
Past versions of "top 10 proteins-1.png"
</button>
</p>
<div id="fig-no-spaces-top-10-proteins-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/9ba78e7f24bd98107a5e62a732537c23398a02c6/docs/figure/Olink-GB.Rmd/top 10 proteins-1.png" target="_blank">9ba78e7</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#alternatively flip the plot horizontally to display more bars </code></pre>
<p>This plot shows the top ten contributing proteins in both PD1 and
PD2. The dashed line at 0.1% represents the average contribution which
is expected in a Principal Component Analysis featuring aproximately
1000 proteins, anything above is considered a contributor.</p>
</div>
<div id="visualizing-contribution-by-protein-in-a-loading-plot"
class="section level3">
<h3>Visualizing contribution by protein in a loading plot</h3>
<pre class="r"><code># Variable Correlation Plot
p_loading &lt;- fviz_pca_var(pca_res, 
                          # Only show the top 20 proteins by &#39;contribution&#39; 
                          # (This keeps the plot clean)
                          select.var = list(contrib = 20), 
                          
                          # Color the proteins by their contribution level
                          col.var = &quot;contrib&quot;, 
                          gradient.cols = c(&quot;navy&quot;, &quot;white&quot;, &quot;indianred&quot;),
                          
                          # Avoid text labels overlapping
                          repel = TRUE, 
                          
                          # Visual aesthetics
                          alpha.var = &quot;contrib&quot;) + 
  theme_minimal() +
  labs(title = &quot;Loading Plot: Top 20 Proteins&quot;,
       subtitle = &quot;Direction shows correlation with PC1 (x) and PC2 (y)&quot;)
p_loading</code></pre>
<p><img src="figure/Olink-GB.Rmd/loading%20plot-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-loading-plot-1">
Past versions of "loading plot-1.png"
</button>
</p>
<div id="fig-no-spaces-loading-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/9ba78e7f24bd98107a5e62a732537c23398a02c6/docs/figure/Olink-GB.Rmd/loading plot-1.png" target="_blank">9ba78e7</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>How to interpret the Loading Plot: 1. Horizontal Arrows: These
proteins are purely PC1 drivers. If they point to the right, they are
high in whatever Group is on the right of your PCA. 2. Vertical Arrows:
These proteins are driving PC2. They don’t differentiate your PC1 groups
much, but they show a separate trend (maybe Age or Sex). 3. Diagonal
Arrows: These proteins contribute to both PC1 and PC2. 4. Clusters: If
you see a “fan” of 5 proteins all pointing in the exact same direction,
those proteins likely belong to the same biological pathway (e.g., a
group of inflammatory cytokines all rising together).</p>
<p>Interpretatio of this Loading Plot: 1. Since all proteins “point” in
the same direction they are all correlated and likely belong to the same
biological process 2. They have a positive correlation with PC1 and a
negative correlation with PC2</p>
</div>
</div>
<div id="differential-expression-analysis"
class="section level2 tabset">
<h2 class="tabset">Differential Expression Analysis</h2>
<p>###Create the Design Matrix</p>
<pre class="r"><code># 1. Clean the &#39;Group&#39; column in your metadata directly
# This replaces hyphens/spaces with dots (e.g., SSc-PAH -&gt; SSc.PAH)
Clinical_data$Group_Clean &lt;- make.names(Clinical_data$Group)

# 2. Re-create the group_factor from the clean column
group_factor &lt;- as.factor(Clinical_data$Group_Clean)

# 3. Build the design matrix
design &lt;- model.matrix(~ 0 + group_factor)

# 4. Rename columns to match the clean levels (SSc.PAH, SSc.noPAH)
colnames(design) &lt;- levels(group_factor)</code></pre>
<div id="define-the-comparison" class="section level3">
<h3>Define the comparison</h3>
<pre class="r"><code># 5. Define the contrast
# Syntax: makeContrasts(Group1 - Group2, levels = design)
contrast.matrix &lt;- makeContrasts(SSc.PAH - SSc.noPAH, levels = design)</code></pre>
</div>
<div id="analysis" class="section level3">
<h3>Analysis</h3>
<pre class="r"><code># Fit the model
fit &lt;- lmFit(Olink_data, design)
fit2 &lt;- contrasts.fit(fit, contrast.matrix)
fit2 &lt;- eBayes(fit2)

# 1. Get the results from limma
de_results &lt;- topTable(fit2, coef = 1, number = Inf, sort.by = &quot;P&quot;) %&gt;%
  rownames_to_column(&quot;Protein&quot;)

# 2. CREATE THE MISSING COLUMN (The Fix)
de_results &lt;- de_results %&gt;%
  mutate(Significance = case_when(
    adj.P.Val &lt; 0.05 &amp; logFC &gt; 0.5  ~ &quot;Up-regulated&quot;,
    adj.P.Val &lt; 0.05 &amp; logFC &lt; -0.5 ~ &quot;Down-regulated&quot;,
    TRUE                          ~ &quot;Not Significant&quot;
  ))</code></pre>
</div>
</div>
<div id="visualization-of-the-differential-analysis"
class="section level2">
<h2>Visualization of the Differential Analysis</h2>
<div id="volcanoplot" class="section level3">
<h3>Volcanoplot</h3>
<pre class="r"><code># --- 1. DATA PREPARATION ---
# Calculate the exact raw p-value that hits the FDR 0.05 threshold
p_cutoff &lt;- de_results %&gt;% 
  filter(adj.P.Val &lt; 0.05) %&gt;% 
  pull(P.Value) %&gt;% 
  max()

# Create a &#39;Score&#39; to identify the &quot;top corners&quot; (High FC * High Significance)
# This ensures we label the most extreme points in the top-left and top-right.
de_results &lt;- de_results %&gt;%
  mutate(label_score = abs(logFC) * -log10(P.Value))

# Identify top 10 Up and top 10 Down based on this combined score
top_up &lt;- de_results %&gt;%
  filter(Significance == &quot;Up-regulated&quot;) %&gt;%
  slice_max(order_by = label_score, n = 10)

top_down &lt;- de_results %&gt;%
  filter(Significance == &quot;Down-regulated&quot;) %&gt;%
  slice_min(order_by = logFC, n = 10) # Using logFC here ensures we stay on the left side

# Re-selecting top_down based on score but ensuring they are actually down-regulated
top_down &lt;- de_results %&gt;%
  filter(Significance == &quot;Down-regulated&quot;) %&gt;%
  slice_max(order_by = label_score, n = 10)

top_labels &lt;- bind_rows(top_up, top_down)</code></pre>
<pre class="r"><code># --- 2. THE FINAL PLOT ---
ggplot(de_results, aes(x = logFC, y = -log10(P.Value), color = Significance)) +
  geom_point(alpha = 0.4, size = 1.5) +
  scale_color_manual(values = c(&quot;Down-regulated&quot; = &quot;navy&quot;, 
                                &quot;Not Significant&quot; = &quot;grey85&quot;, 
                                &quot;Up-regulated&quot; = &quot;indianred&quot;)) +
  
  # Clean, small text labels
  geom_text_repel(data = top_labels, 
                  aes(label = Protein),
                  size = 2.5,
                  color = &quot;black&quot;,
                  box.padding = 0.5,
                  point.padding = 0.2,
                  min.segment.length = 0,
                  segment.size = 0.2,
                  segment.color = &quot;grey30&quot;,
                  max.overlaps = Inf) +
  
  theme_minimal() +
  labs(title = &quot;SSc.PAH vs SSc.noPAH: Top Hits&quot;,
       subtitle = &quot;Labeled by multiplied Significance and Fold Change&quot;,
       x = &quot;Log2 Fold Change&quot;,
       y = &quot;-log10(Raw P-value)&quot;) +
  
  # Guidelines
  geom_vline(xintercept = c(-0.5, 0.5), linetype = &quot;dotted&quot;, color = &quot;grey40&quot;) +
  geom_hline(yintercept = -log10(p_cutoff), linetype = &quot;dashed&quot;, color = &quot;grey40&quot;, alpha = 0.7) +
  
  # Move FDR Label to the far LEFT to avoid crashing with the &quot;Up-regulated&quot; top corner
  annotate(&quot;label&quot;, 
           x = min(de_results$logFC), 
           y = -log10(p_cutoff), 
           label = &quot;FDR &lt; 0.05&quot;, 
           hjust = 0.2,             # Align to the left of the x-coordinate
           vjust = 1.4, 
           size = 2.5, 
           color = &quot;black&quot;, 
           fill = &quot;white&quot;,        # White background prevents &quot;crashing&quot; with grid lines
           label.size = NA,       # Removes the box border
           fontface = &quot;plain&quot;)</code></pre>
<p><img src="figure/Olink-GB.Rmd/volcano%20plotting-1.png" width="2100" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-volcano-plotting-1">
Past versions of "volcano plotting-1.png"
</button>
</p>
<div id="fig-no-spaces-volcano-plotting-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexander-gysin/omics-training/blob/9ba78e7f24bd98107a5e62a732537c23398a02c6/docs/figure/Olink-GB.Rmd/volcano plotting-1.png" target="_blank">9ba78e7</a>
</td>
<td>
alexander-gysin
</td>
<td>
2026-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>continue with: 1. The Heatmap (Sample Clustering) A heatmap is
essential because it tells you if your “top hits” actually succeed in
grouping your patients correctly. If the SSc-PAH patients don’t cluster
together, your differential proteins might be driven by outliers.</p>
<ol start="2" style="list-style-type: decimal">
<li><p>Boxplots (The “Deep Dive”) Once you’ve identified your #1 protein
(e.g., the one with the highest “score” from our last plot), you need to
see the raw data distribution. A boxplot shows you the spread, the
median, and if there are any outliers.</p></li>
<li><p>MA Plot (The “Sanity Check”) An MA plot looks at the relationship
between the average expression (A) and the log fold change (M). Purpose:
It helps you check if your significant results only happen in very
low-abundance proteins (which can be noisy) or high-abundance ones.
Ideally, your “red dots” should be spread across the x-axis.</p></li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.5.2 (2025-10-31)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 24.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] patchwork_1.3.2        pheatmap_1.0.13        EnhancedVolcano_1.28.2
 [4] ggrepel_0.9.6          factoextra_1.0.7       FactoMineR_2.13       
 [7] limma_3.66.0           readxl_1.4.5           RColorBrewer_1.1-3    
[10] magrittr_2.0.4         here_1.0.2             lubridate_1.9.4       
[13] forcats_1.0.1          stringr_1.5.2          dplyr_1.1.4           
[16] purrr_1.1.0            readr_2.1.5            tidyr_1.3.1           
[19] tibble_3.3.0           ggplot2_4.0.0          tidyverse_2.0.0       
[22] workflowr_1.7.2       

loaded via a namespace (and not attached):
 [1] sandwich_3.1-1       rlang_1.1.6          git2r_0.36.2        
 [4] multcomp_1.4-29      otel_0.2.0           compiler_4.5.2      
 [7] getPass_0.2-4        callr_3.7.6          vctrs_0.6.5         
[10] pkgconfig_2.0.3      fastmap_1.2.0        backports_1.5.0     
[13] labeling_0.4.3       promises_1.4.0       rmarkdown_2.30      
[16] tzdb_0.5.0           ps_1.9.1             xfun_0.53           
[19] cachem_1.1.0         jsonlite_2.0.0       flashClust_1.01-2   
[22] later_1.4.4          broom_1.0.10         cluster_2.1.8.1     
[25] R6_2.6.1             bslib_0.9.0          stringi_1.8.7       
[28] car_3.1-5            jquerylib_0.1.4      cellranger_1.1.0    
[31] estimability_1.5.1   Rcpp_1.1.0           knitr_1.50          
[34] zoo_1.8-14           httpuv_1.6.16        Matrix_1.7-4        
[37] splines_4.5.2        timechange_0.3.0     tidyselect_1.2.1    
[40] rstudioapi_0.17.1    abind_1.4-8          yaml_2.3.10         
[43] codetools_0.2-20     processx_3.8.6       lattice_0.22-7      
[46] withr_3.0.2          S7_0.2.0             coda_0.19-4.1       
[49] evaluate_1.0.5       survival_3.8-3       pillar_1.11.1       
[52] ggpubr_0.6.2         carData_3.0-6        whisker_0.4.1       
[55] DT_0.34.0            generics_0.1.4       rprojroot_2.1.1     
[58] hms_1.1.4            scales_1.4.0         xtable_1.8-4        
[61] leaps_3.2            glue_1.8.0           emmeans_2.0.1       
[64] scatterplot3d_0.3-44 tools_4.5.2          ggsignif_0.6.4      
[67] fs_1.6.6             mvtnorm_1.3-3        grid_4.5.2          
[70] Formula_1.2-5        cli_3.6.5            gtable_0.3.6        
[73] rstatix_0.7.3        sass_0.4.10          digest_0.6.37       
[76] TH.data_1.1-4        htmlwidgets_1.6.4    farver_2.1.2        
[79] htmltools_0.5.8.1    lifecycle_1.0.4      httr_1.4.7          
[82] multcompView_0.1-11  statmod_1.5.1        MASS_7.3-65         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
